hazelcast/src/main/java/com/hazelcast/query/impl/getters/AbstractJsonGetter.java

```diff
@@ -21,13 +21,43 @@
 import com.fasterxml.jackson.core.JsonTokenId;
 import com.hazelcast.core.HazelcastException;
 import com.hazelcast.internal.json.JsonValue;
+import com.hazelcast.internal.serialization.impl.NavigableJsonInputAdapter;
+import com.hazelcast.json.internal.JsonPattern;
+import com.hazelcast.json.internal.JsonSchemaHelper;
+import com.hazelcast.json.internal.JsonSchemaNode;
+import com.hazelcast.util.collection.WeightedEvictableList.WeightedItem;
 
 import java.io.IOException;
+import java.util.List;
 
 public abstract class AbstractJsonGetter extends Getter {
 
+    /**
+     * The number of times this getter will try to use previously observed
+     * patterns. The getter tries most commonly observed patterns first.
+     * If known patterns do not work, then a new pattern is created for
+     * this object.
+     */
+    private static final String PATTERN_TRY_COUNT_PROPERTY = "com.hazelcast.internal.pattern.try.count";
```

Comment:
This should be at least a group property or even better... a configuration option. JVM args can be a pain when running multiple instances in the same JVM. If you think this is not going to be useful for the users, maybe you might get away with a group property. If you would like to avoid the whole config ceremony, maybe you could hardcode this to a reasonable default.

Reply:
I think we should hardcode a reasonable default. I have put the property to easily test different values. I'll replace this with a default before merging.